internal/rpcserver/rpcserver.go

```diff
@@ -1080,8 +1080,23 @@ func handleDebugLevel(_ context.Context, s *Server, cmd interface{}) (interface{
 // createVinList returns a slice of JSON objects for the inputs of the passed
 // transaction.
 func createVinList(mtx *wire.MsgTx, isTreasuryEnabled bool) []types.Vin {
-	// Coinbase transactions only have a single txin by definition.
+	// Treasurybase transactions only have a single txin by definition.
+	//
+	// NOTE: This check MUST come before the coinbase check because a
+	// treasurybase will be identified as a coinbase as well.
```

Comment:
From what I can tell `standalone.IsCoinBaseTx` will not identify treasurybase as coinbase *if* `isTreasuryEnabled` is accurate because of the treasury-like checks in `IsCoinBaseTx` (the `isTreasurySpendLike` closure), but I think it's prudent to check treasurybase first regardless.

Reply:
It will.  It won't identify treasury spends as coinbases, but it will identify a treasurybase as a coinbase.

Reply:
Ah, *spends*.  Thanks